### ISSUE TITLE
pocean-core 2.0.0 requires an older version of pandas

### ISSUE THREAD

**Issue by eldobbins** [pyoceans/pocean-core#62]
Hi;

I used mamba (conda) to install pocean-core from the conda-forge channel into an empty conda environment.  
``` > mamba install pocean-core gsw -c conda-forge ```
That also installed pandas 0.25.3
``` 
...
pandas            0.25.3  py38h4f17bb1_0          conda-forge/osx-64      11 MB 
pocean-core        2.0.0  pyh9f0ad1d_0            conda-forge/noarch       4 MB
...
```

I would like to upgrade to the most recent version of pandas, but when I try, mamba wants to revert pocean-core to v1.9.3
```
(ctd_2021) eld-iMac:~ eldobbins$ mamba install pandas=1.2.4
...
pkgs/main/osx-64         Using cache
pkgs/main/noarch         Using cache
pkgs/r/osx-64            Using cache
pkgs/r/noarch            Using cache
conda-forge/osx-64       Using cache
conda-forge/noarch       Using cache

Looking for: ['pandas=1.2.4']

Transaction

  Prefix: /Users/eldobbins/anaconda3/envs/ctd_2021

  Updating specs:

   - pandas==1.2.4

  Package      Version  Build           Channel                Size
  Upgrade:
  pandas        0.25.3  py38h4f17bb1_0  installed                  
  pandas         1.2.4  py38hb2f4e1b_0  pkgs/main/osx-64       8 MB

  Downgrade:
  pocean-core    2.0.0  pyh9f0ad1d_0    installed                  
  pocean-core    1.9.3  py_0            conda-forge/noarch     4 MB

  Summary:

  Upgrade: 1 packages
  Downgrade: 1 packages

  Total download: 13 MB
```

Is there a way to keep the most recent versions of both pandas and pocean-core?

PS. If there is some testing/dev of pocean-core that needs to happen to enable this, I'm willing to help do it. Just let me know what to do.

**Comment by ocefpaf** [comment]
The feedstock explicitly pins to older pandas in https://github.com/conda-forge/pocean-core-feedstock/blob/master/recipe/meta.yaml#L24 and that seems to agree with the [requirements in the package](https://github.com/pyoceans/pocean-core/blob/master/requirements.txt#L3).

I'll send a PR to move the tests from Travis-CI and AppVeyor to GitHub Actions, after that I'll test why the new pandas is a blocker here and hopefully, fix it.

**Comment by kwilcox** [comment]
It is only used/testesd against `pandas<1.0`, you can bump the requirements locally, run the tests, and see what breaks 😛

**Comment by ocefpaf** [comment]
> It is only used/testesd against `pandas<1.0`, you can bump the requirements locally, run the tests, and see what breaks

That's the plan :smile:

**Comment by eldobbins** [comment]
Before I tried pandas 1.2.4, I tried running the tests with pandas 0.25.3.  I had to restrict numpy to <1.20 or I got a dozen dtype errors related to https://github.com/numpy/numpy/issues/18355.  But even with a lower version of numpy, I am still failing 3 tests.  

Trajectory tests are giving me:
```
pocean/tests/dsg/trajectory/test_trajectory_im.py:146: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pocean/dsg/trajectory/im.py:179: in from_dataframe
    times = get_ncdata_from_series(gdf[axes.t], time)
pocean/utils.py:314: in get_ncdata_from_series
    nums = date2num(series.tolist(), units=units, calendar=calendar)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   TypeError: unsupported operand type(s) for %: 'NaTType' and 'datetime.timedelta'
```

and test_timeseries_om.py is having trouble with the fill value:
```
>                   fill_value = np.array(fill_value, copy=False, dtype=ndtype)
E                   OverflowError: Python int too large to convert to C long

../../../anaconda3/envs/pocean/lib/python3.7/site-packages/numpy/ma/core.py:468: OverflowError
```

I tried pandas 1.2.4 and got these same errors, so I'm going to try and track them down.

**Comment by eldobbins** [comment]
My problem was a still a numpy-pandas mis-match.

This passed all tests:
```
numpy                     1.15.4           py37hacdab7b_0
pandas                    0.25.3           py37h0a44026_0  
```

**Comment by ocefpaf** [comment]
Yep. See https://github.com/numpy/numpy/issues/18355
Ideally we should fix pocean core to work with both updated versions of numpy and pandas. See #63 for my attempted to fix it.

**Comment by kwilcox** [comment]
PR is now https://github.com/pyoceans/pocean-core/pull/66 because... well... :running_man:

**Comment by kwilcox** [comment]
@eldobbins next release will include support for `pandas>1.0.5`, thanks for the persistence!
@ocefpaf thanks for the solution!

**Comment by eldobbins** [comment]
I just updated my code.  Pandas 1.2.4 is passing all tests on my older Mac (High Sierra, MacOS 10.13.6)!  Thank you.

Oddly, on my newer Mac laptop (Mojave, 10.14.6) I am getting 3 errors.  I think you might be also, given the commit message on requirements.txt.  The error message is:

> ValueError: Big-endian buffer not supported on little-endian compiler
> pandas/_libs/algos_take_helper.pxi

If that isn't the error you are seeing, could you let me know?  I googled that error and came up with a [thread on the Astropy GitHub](https://github.com/astropy/astropy/issues/11286) that seems relevant.  If it is my problem alone, I'll try their fix.

**Comment by kwilcox** [comment]
I've hit this a few times over the years, it has to do with the computer that wrote the data having a different endiness than the computer reading it. Will require a full byteswap on the data. Curious - Is your newer Mac using an M1 (big endian)? 

https://numpy.org/devdocs/user/basics.byteswapping.html

**Comment by joefutrelle** [comment]
Very timely fix for me as one of my other dependencies requires Pandas >= 1.0. @kwilcox what's the timeline for the release?

**Comment by joefutrelle** [comment]
Bumping this issue in hopes that there will be a release that contains the most recent fixes.

**Comment by kwilcox** [comment]
Trying to make a pocean release but there is some sort of naming conflict in PyPi with the package name. @ocefpaf Do you happen to have a `pocean-core` project on https://pypi.org/ with no releases? Uploading to https://test.pypi.org/ works fine!

**Comment by ocefpaf** [comment]
> Trying to make a pocean release but there is some sort of naming conflict in PyPi with the package name. @ocefpaf Do you happen to have a `pocean-core` project on https://pypi.org/ with no releases? Uploading to https://test.pypi.org/ works fine!

See https://github.com/pypa/pypi-support/issues/1752 for more info on that error. The name should be release to you soon.

**Comment by ocefpaf** [comment]
I believe this issue can be closed. We issue released 3.0.0 a while back and that one works with latest pandas. Please re-open if I'm missing something that is not fixed yet.
